src/App/Transactions.h

```diff
@@ -98,73 +160,147 @@ class AppExport Transaction: public Base::Persistence
+ * that is a base class that contains functionality for a DocumentObject
+ * regarding transactions.
  */
 class AppExport TransactionObject: public Base::Persistence
 {
     TYPESYSTEM_HEADER_WITH_OVERRIDE();
 
 public:
-    /// Construction
+    /// Construct a transaction object.
     TransactionObject();
-    /// Destruction
+
+    /// Destruct a transaction object.
     ~TransactionObject() override;
 
-    virtual void applyNew(Document& Doc, TransactionalObject* pcObj);
-    virtual void applyDel(Document& Doc, TransactionalObject* pcObj);
-    virtual void applyChn(Document& Doc, TransactionalObject* pcObj, bool Forward);
+    /**
+     * @brief Apply the transaction that adds a new object to the document.
+     *
+     * @param[in,out] doc The document to apply the transaction to.
+     * @param[in,out] obj The object that is added in the transaction.
+     */
+    virtual void applyNew(Document& doc, TransactionalObject* obj);
+
+    /**
+     * @brief Apply the transaction that removes an object from the document.
+     *
+     * @param[in,out] doc The document to apply the transaction to.
+     * @param[in,out] obj The object that is removed in the transaction.
+     */
+    virtual void applyDel(Document& doc, TransactionalObject* obj);
+
+    /**
+     * @brief Apply the transaction that changes an object in the document.
+     *
+     * @param[in,out] doc The document to apply the transaction to.
+     * @param[in,out] obj The object that is changed in the transaction.
+     * @param[in] forward If true, apply the transaction; otherwise, undo it.
+     */
+    virtual void applyChn(Document& doc, TransactionalObject* obj, bool forward);
 
-    void setProperty(const Property* pcProp);
-    void addOrRemoveProperty(const Property* pcProp, bool add);
+    /**
+     * @brief Set the property of the object that is affected by the transaction.
+     *
+     * @param[in] prop The property that is affected by the transaction.
+     */
+    void setProperty(const Property* prop);
+
+    /**
+     * @brief Add or remove a property from the object.
+     *
+     * @param[in] prop The property to add or remove.
+     * @param[in] add If true, add the property; otherwise, remove it.
+     */
+    void addOrRemoveProperty(const Property* prop, bool add);
 
     unsigned int getMemSize() const override;
     void Save(Base::Writer& writer) const override;
-    /// This method is used to restore properties from an XML document.
     void Restore(Base::XMLReader& reader) override;
 
+    /// Give Transaction access to the internals of this class.
     friend class Transaction;
 
 protected:
+    /// The status of the transaction object.
     enum Status
     {
+        /// A new object is added to the document.
         New,
+        /// An object is removed from the document.
         Del,
+        /// An object is changed in the document.
         Chn
     } status {New};

```

Comment:
I think this is a good cases where [`///<`](https://www.doxygen.nl/manual/docblocks.html#memberdoc) would be useful
```suggestion
    enum Status
    {
        New, ///< A new object is added to the document.
        Del, ///< An object is removed from the document.
        Chn ///< An object is changed in the document.
    } status {New};
```